### PR TITLE
utl: add tee TCL command

### PR DESCRIPTION
**Pull request by gadfort**
Adds:
- `tee` as a tcl command modeled after: https://yosyshq.readthedocs.io/projects/yosys/en/latest/cmd/tee.html

### CODE REVIEW

**Comment by maliberty** [comment]
This seems to make the previous PR for redirection redundant as `tee -quiet` does the same.

File: src/utl/README.md [hunk 1]
```diff
@@ -51,6 +51,33 @@ man clock_tree_synthesis
...
+    [-append]
+    [-quiet]
+    command
+```
+
+#### Options
+
+| Switch Name | Description | 
+| ----- | ----- |
+| `filename` | File to redirect output into. |
```
Comment (maliberty) [inline src/utl/README.md:69]: -file filename

File: src/utl/src/Logger.cpp [hunk 2]
```diff
@@ -239,6 +239,24 @@ void Logger::redirectFileEnd()
...
+void Logger::teeFileBegin(const std::string& filename)
+{
+  file_redirect_ = std::make_unique<std::ofstream>(filename.c_str());
+  setRedirectSink(*file_redirect_, true);
+}
+
+void Logger::teeFileAppendBegin(const std::string& filename)
+{
+  file_redirect_
+      = std::make_unique<std::ofstream>(filename.c_str(), std::ofstream::app);
```
Comment (maliberty) [inline src/utl/src/Logger.cpp:251]: `.c_str()` is unnecessary

**Comment by maliberty** [comment]
I know its a bit odd but what happens with `tee -file foo { report_checks > bar }`?

**Review by maliberty — CHANGES_REQUESTED** [review]

**Comment by gadfort** [comment]
> This seems to make the previous PR for redirection redundant as `tee -quiet` does the same.

@maliberty yes, this provides a consistent interface to use.


> I know its a bit odd but what happens with `tee -file foo { report_checks > bar }`?

I don't think we want to support that, so it should throw an error.

**Comment by maliberty** [comment]
> should throw an error.

Does it?

**Review by maliberty — APPROVED** [review]

**Pull request merged**